src/geocoder.js

```diff
@@ -35,12 +40,16 @@ module.exports = function(url, options) {
         }
 
         feedback.record({geocoding: query});
-        return L.Util.template(geocoder.getURL(), {query: query});
+
+        return L.Util.template(geocoder.getURL(), {
+            proximity: args.proximity ? encodeURIComponent(args.proximity[0]+','+args.proximity[1]) : false,
+            query: query 
+        });
     };
 
-    geocoder.query = function(_, callback) {
+    geocoder.query = function(_, args, callback) {
```

Comment:
[`query` is a public API](https://www.mapbox.com/mapbox.js/api/v2.1.9/l-mapbox-geocoder/#section-geocoder-query) -- we can't make a backwards incompatible change to the method signature until 3.0.

We could make the first parameter accept an object as well as a plain string, and support:

`geocoder.query({query: 'austin', proximity: latlng}, callback)`

WDYT?